sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java

```diff
@@ -127,7 +127,29 @@ public BlobClientBuilder() {
      * and {@link #retryOptions(RequestRetryOptions)} have been set.
      */
     public BlobClient buildClient() {
-        return new BlobClient(buildAsyncClient());
+        Objects.requireNonNull(blobName, "'blobName' cannot be null.");
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null");
+
+        BuilderHelper.httpsValidation(customerProvidedKey, "customer provided key", endpoint, LOGGER);
+
+        if (Objects.nonNull(customerProvidedKey) && Objects.nonNull(encryptionScope)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("Customer provided key and encryption "
+                + "scope cannot both be set"));
+        }
+
+        /*
+        Implicit and explicit root container access are functionally equivalent, but explicit references are easier
+        to read and debug.
+         */
+        String blobContainerName = CoreUtils.isNullOrEmpty(containerName) ? BlobContainerClient.ROOT_CONTAINER_NAME
+            : containerName;
+
+        BlobServiceVersion serviceVersion = version != null ? version : BlobServiceVersion.getLatest();
+
+        BlobAsyncClient asyncClient = buildAsyncClient();
+
+        return new BlobClient(asyncClient, asyncClient.getHttpPipeline(), endpoint, serviceVersion, accountName,
```

Comment:
This is to ensure that there aren't two different http pipelines being used, since the async client is still being used for the upload and download operations. We'd want to keep this until we completely decouple the sync client from the async.

Reply:
Makes sense. Thanks for clarifying.